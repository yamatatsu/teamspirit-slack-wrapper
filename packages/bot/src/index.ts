import {
  puppeteer,
  args,
  defaultViewport,
  executablePath,
  headless,
} from "chrome-aws-lambda"

export const handler = async () => {
  let result = null
  let browser = null

  try {
    browser = await puppeteer.launch({
      args,
      defaultViewport,
      executablePath: await executablePath,
      headless,
      ignoreHTTPSErrors: true,
    })

    let page = await browser.newPage()

    await page.goto("https://example.com")

    const bodyHTML = await page.content()
    console.log(bodyHTML)

    return result
  } finally {
    if (browser !== null) {
      await browser.close()
    }
  }
}
