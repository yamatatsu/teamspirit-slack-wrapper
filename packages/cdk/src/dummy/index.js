exports.handler = async (event) => {
  console.info(event)
  return event
}
