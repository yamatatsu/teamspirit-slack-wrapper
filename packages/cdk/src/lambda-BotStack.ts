import * as cdk from "@aws-cdk/core"
import * as lambda from "@aws-cdk/aws-lambda"

type Props = cdk.StackProps & {
  code: lambda.Code
}

export class BotStack extends cdk.Stack {
  readonly lambda: lambda.IFunction
  constructor(parent: cdk.Construct, id: string, props: Props) {
    super(parent, id, props)

    const { code } = props

    this.lambda = new lambda.Function(this, "lambdaFunction", {
      code,
      handler: "index.handler",
      runtime: lambda.Runtime.NODEJS_12_X,
      tracing: lambda.Tracing.ACTIVE,
    })
  }
}
