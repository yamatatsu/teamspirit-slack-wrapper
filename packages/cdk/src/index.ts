import * as cdk from "@aws-cdk/core"
import * as lambda from "@aws-cdk/aws-lambda"
import { BotStack } from "./lambda-BotStack"

const env = { region: "ap-northeast-1" }

const app = new cdk.App()

const code = new lambda.AssetCode(`${__dirname}/dummy`)
new BotStack(app, "BotStack", { code, env })
