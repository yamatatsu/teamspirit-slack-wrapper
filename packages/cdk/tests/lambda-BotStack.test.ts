import { SynthUtils } from "@aws-cdk/assert"
import { Stack } from "@aws-cdk/core"
import { AssetCode } from "@aws-cdk/aws-lambda"
import { BotStack } from "../src/lambda-BotStack"

test("BotStack snapshot test", () => {
  const stack = new Stack()
  const env = { region: "ap-northeast-1", account: "xxxxxs" }

  const target = new BotStack(stack, "testBotStack", {
    code: new AssetCode(`${__dirname}/assets`),
    env,
  })

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot()
})
