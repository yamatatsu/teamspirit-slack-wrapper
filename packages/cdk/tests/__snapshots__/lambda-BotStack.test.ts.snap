// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotStack snapshot test 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersbd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031eArtifactHashFC4A34F8": Object {
      "Description": "Artifact hash for asset \\"bd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031e\\"",
      "Type": "String",
    },
    "AssetParametersbd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031eS3Bucket0AE755F6": Object {
      "Description": "S3 bucket for asset \\"bd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031e\\"",
      "Type": "String",
    },
    "AssetParametersbd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031eS3VersionKey5802EF7E": Object {
      "Description": "S3 key for asset version \\"bd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031e\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "lambdaFunction940E68AD": Object {
      "DependsOn": Array [
        "lambdaFunctionServiceRoleDefaultPolicy0958DD5A",
        "lambdaFunctionServiceRoleF7E54F17",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersbd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031eS3Bucket0AE755F6",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersbd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031eS3VersionKey5802EF7E",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersbd2c14e17e788a4d2e39a4f6550c415bff26eeb7e6a8365355441fcad7e8031eS3VersionKey5802EF7E",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "MemorySize": 1600,
        "Role": Object {
          "Fn::GetAtt": Array [
            "lambdaFunctionServiceRoleF7E54F17",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 30,
        "TracingConfig": Object {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "lambdaFunctionServiceRoleDefaultPolicy0958DD5A": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lambdaFunctionServiceRoleDefaultPolicy0958DD5A",
        "Roles": Array [
          Object {
            "Ref": "lambdaFunctionServiceRoleF7E54F17",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "lambdaFunctionServiceRoleF7E54F17": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
